### PR TITLE
wsd: exclude readonly sessions from modification indication

### PR DESCRIPTION
**Pull request by Ashod**
Read only sessions cannot modify the document, so
we shouldn't consider input from such a session
as potentially modifying the document. This
becomes important during an abrupt disconnection
where we look at isPossiblyModified() as an
indicator for data-loss.

New unit-test that reproduces the corner-case
with disconnection. With graceful unloading
a previous unit-test verified that it worked
as expected.

Change-Id: Id84cda4f4599c559018247c32ea1205e154e4984
Signed-off-by: Ashod Nakashian <ashod.nakashian@collabora.co.uk>

### CODE REVIEW

File: wsd/ClientSession.cpp [hunk 1]
```diff
@@ -799,7 +799,8 @@ bool ClientSession::_handleInput(const char *buffer, int length)
             }
             else
             {
-                docBroker->updateLastModifyingActivityTime();
+                if (!isReadOnly())
+                    docBroker->updateLastModifyingActivityTime();
```
Comment (eszkadev) [inline wsd/ClientSession.cpp:803]: "moveselectedclientparts: Moves the selected parts to a new position."

should be that possible in readonly mode?

**Review by eszkadev — APPROVED** [review]
Looks ok

File: wsd/ClientSession.cpp (hunk 1 above)
Comment (Ashod) [reply wsd/ClientSession.cpp:803]: I expect that would modify the document and there not allowed.

**Comment by timar** [comment]
@Ashod you did not think about the PDF case, apparently. PDF is read-only but comments can be added and saved.
